presto-main/src/main/java/com/facebook/presto/operator/PartitionedConsumption.java

```diff
@@ -78,7 +85,7 @@ public PartitionedConsumption(
         for (Integer partitionNumber : partitionNumbers) {
             Partition<T> partition = new Partition<>(consumersCount, partitionNumber, loader, partitionActivator, disposer);
             partitions.add(partition);
-            partitionActivator = partition.released;
+            partitionActivator = disposed.apply(partitionNumber);
```

Comment:
Before the fix:  partition is released  (next partition is ready to load) -> signal is sent to operator -> operator dispose
After the fix:  partition is released -> signal is sent to operator -> operator dispose started and completed ( (next partition is ready to load) )